content/waku/2024-gantt.md

```diff
@@ -24,92 +24,128 @@ gantt
+        PostgreSQL maintenance: crit, milestone, after pgsql-m, 0
     section Direct Message Reliability
-        (testing) direct messages: milestone, after test-direct-msg, 0
-        Review connection management: crit, milestone, after review-conn-mgmt-1 review-conn-mgmt-2, 0
-        (tooling) filter and light push protocols: milestone, after lite-proto-tester, 0
-        (telemetry) Fleet logging: milestone, after telem-fleet-logging, 0
-        (telemetry) direct message reliability: milestone, after telem-d-msg-rel, 0
-        Reliability Protocol for Relay: crit, milestone, after rel-relay-g rel-relay-n, 0
-        Reliability Protocol for Resource-Restricted Clients: crit, milestone, after rel-reqres-g rel-reqres-j, 0
+        (telemetry) direct message reliability: milestone, after telem-d-msg-rel-1 telem-d-msg-rel-2, 0
+        Reliability Protocol for Resource-Restricted Clients: crit, milestone, after rel-reqres-j-1 rel-reqres-j-2, 0
         Review MVDS usage and fail path: crit, milestone, after mvds, 0
-        TBC User apps for large scale dogfooding: milestone, after user-app-gui, 0 
+        PostgreSQL optimisation - phase 1: milestone, after pgsql-opt-1, 0
     section E2e reliability protocol
         (telemetry) Multicast message reliability: milestone, after telem-m-msg-rel, 0
         E2e reliability protocol PoC: milestone, crit, after e2e-rel-r, 0
-        E2e reliability protocol Status integration: milestone, crit, after e2e-rel-g, 0
+        E2e reliability protocol Status integration: milestone, crit, after e2e-rel-status-r e2e-rel-status-g, 0
     section Static Sharding - dedicated shards
         (telemetry) Measure Bandwidth: milestone, after telem-bandwidth, 0
-        Sharding peer mgmt and discovery hardening: crit, milestone, after sh-peer-mgmt-j sh-peer-mgmt-n sh-peer-mgmt-g, 0
+        Sharding peer mgmt and discovery hardening: crit, milestone, after sh-peer-mgmt-n sh-peer-mgmt-g, 0
         (testing) Custom shard impl of Communities: milestone, after test-custom-shard, 0
+        PostgreSQL optimisation - phase 2: milestone, after pgsql-opt-2, 0
+    section Bandwidth optimisation and protocol review
+        %% TODO: deliverables are being re-worked
+        Specify de-MLS over Waku: milestone, after demls-spec, 0
+        Review usage of content topics in Status Chat and Communities protocol: milestone, after cont-topic-r, 0
+        Minimal Community Specification and Implementation: milestone, after min-com-spec, 0
+        %% TODO: schedule
+        Telemetry review: milestone, after telem-review, 0
+        Minimal solution for greedy messages: milestone, after min-sol-com, 0
+        Define long-term solution: milestone, after define-com-sol, 0
+    section Nwaku in Status Desktop (Relay mode)
+        Nwaku on Windows: milestone, after nwaku-windows-n, 0
+        Nwaku in Status Desktop: milestone, after nwaku-status-desktop, 0
```

Comment:
@Ivansete-status @richard-ramos please assist with an estimate on those.

Reply:
> @Ivansete-status @richard-ramos please assist with an estimate on those.

I'm a bit confused about the multiple `nwaku-status-desktop` milestones.
On the other hand, some very initial steps we've taken in that direction aimed to first bring `nwaku` in `status-go`. Then I think it is interesting to consider the `nwaku-in-status-go` milestone as a pre-requisite of `nwaku-status-desktop`.

That `nwaku-in-status-go` should achieve, at least:
- interconnect status-go-CLI with nwaku
- make `wakuv2/waku_test.go` run properly while linking `nwaku`/`libwaku`

Reply:
> I'm a bit confused about the multiple nwaku-status-desktop milestones.

Only 1 milestone and 2 deliverables:

```
    section Nwaku in Status Desktop (Relay mode)
        Nwaku on Windows: milestone, after nwaku-windows-n, 0
        Nwaku in Status Desktop: milestone, after nwaku-status-desktop, 0
```

I recently simplified that, the roadmap Google doc has the same split now: https://docs.google.com/document/d/1L8HvXtAYk-JqQL6w3RgCskXwegcTa0J5nyH9YL4LrQE/edit#heading=h.ic3gjyufc43j

Based on that, please provide an estimate :)

Reply:
@fryorcraken - after a conversation with @richard-ramos I think we can consider the end of January'25 to have this completed. Richard correct me if you don't agree.